src/main/java/seedu/address/model/task/DeadlineTask.java

```diff
@@ -3,28 +3,49 @@
 import java.time.LocalDateTime;
 import java.util.Objects;
 
+import java.util.function.Predicate;
+
 import seedu.address.commons.exceptions.IllegalValueException;
 
 public class DeadlineTask extends Task {
 
-    private static final String FMT_STRING = "DeadlineTask[name=%s, due=%s]";
+    private static final String FMT_STRING = "DeadlineTask[name=%s, due=%s, finished=%s]";
 
     private final LocalDateTime due;
 
-    public DeadlineTask(Name name, LocalDateTime due) {
+    private boolean finished;
+
+    public static Predicate<? super DeadlineTask> isNotFinishedDeadline() {
+        return p -> !p.isFinished();
+    }
```

Comment:
Is there any reason why this has to be a part of `DeadlineTask`? Don't we already have the `seedu.address.model.filter` package for that putting all the model filters inside?